### PR TITLE
Reuse HTTP client instance

### PR DESCRIPTION
**Pull request by sargun**

### CODE REVIEW

**Comment by sargun** [comment]
@kaysoky Mind running a build of this, and seeing if it solves the problem?

File: records/generator.go [hunk 1]
```diff
@@ -226,6 +225,7 @@ func (rg *RecordGenerator) loadFromMaster(ip string, port string) (state.State,
 	}
 
 	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("User-Agent", "Mesos-DNS")
```
Comment (jdef) [inline records/generator.go]: :+1:

File: resolver/resolver.go [hunk 2]
```diff
@@ -25,27 +25,47 @@ import (
...
+	version    string
+	config     records.Config
+	rs         *records.RecordGenerator
+	rsLock     sync.RWMutex
+	rng        *rand.Rand
+	fwd        exchanger.Forwarder
+	httpClient *http.Client
+}
+
+func newHTTPClient(httpTimeout time.Duration) *http.Client {
```
Comment (jdef) [inline resolver/resolver.go]: probably need to sync w/ @alberts

Comment (sargun) [inline resolver/resolver.go]: Talked to @spahl. He thinks our changes make sense, and he's going to wait till we merge to master (soon), and then he can rebase atop us.

**Comment by sargun** [comment]
@jdef Looks good otherwise? The only thing I was thinking of is adding the PID to the user agent. What do you think?

**Comment by jdef** [comment]
@sargun LGTM; no pid needed (too leaky)

**Pull request merged**